src/Shared/DiagnosticSourceInstrumentation/PropertyFetcher.cs

```diff
@@ -105,10 +93,28 @@ static PropertyFetch CreateFetcherForProperty(PropertyInfo propertyInfo)
                     return null;
                 }
 
-                var typedPropertyFetcher = typeof(TypedPropertyFetch<,>);
-                var instantiatedTypedPropertyFetcher = typedPropertyFetcher.MakeGenericType(
-                    typeof(T), propertyInfo.DeclaringType, propertyInfo.PropertyType);
-                return (PropertyFetch)Activator.CreateInstance(instantiatedTypedPropertyFetcher, propertyInfo);
+                if (propertyInfo.DeclaringType!.IsValueType || propertyInfo.PropertyType.IsValueType || typeof(T).IsValueType)
+                {
+                    return new BoxedValueTypePropertyFetch(propertyInfo);
```

Comment:
@Yun-Ting Does any of the existing code using `PropertyFetcher` today hit this branch? If not, could we just change this into a `throw NotSupportedException` and mark it all as AoT-safe?

Reply:
Unfortunately it is used: https://github.com/open-telemetry/opentelemetry-dotnet/blob/0343116e7781779975a1a0fe269be1778a484eeb/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs#L46

The `TaskStatus` is a value type (enum) and so we can't use the `MakeGenericType` in AOT for it.